lib/options.ts

```diff
@@ -48,42 +48,59 @@ export type UploadInput =
   // available in React Native
   | ReactNativeFile
 
-export interface UploadOptions {
-  endpoint?: string
+/**
+ * Options for configuring stall detection behavior
+ */
+export interface StallDetectionOptions {
+  enabled: boolean
+  stallTimeout: number // Time in ms before considering progress stalled
+  checkInterval: number // How often to check for stalls
+}
 
-  uploadUrl?: string
-  metadata: { [key: string]: string }
-  metadataForPartialUploads: UploadOptions['metadata']
-  fingerprint: (file: UploadInput, options: UploadOptions) => Promise<string | null>
-  uploadSize?: number
+export type Part = { start: number; end: number }
 
-  onProgress?: (bytesSent: number, bytesTotal: number | null) => void
-  onChunkComplete?: (chunkSize: number, bytesAccepted: number, bytesTotal: number | null) => void
-  onSuccess?: (payload: OnSuccessPayload) => void
-  onError?: (error: Error | DetailedError) => void
-  onShouldRetry?: (error: DetailedError, retryAttempt: number, options: UploadOptions) => boolean
-  onUploadUrlAvailable?: () => void | Promise<void>
+export interface UploadOptions {
+  endpoint?: string | null
```

Comment:
A lot of the changes in this file seem unrelated to the PR, such as these type changes. Can you please remove them (or propose them in a separate PR) to keep this one focused?